JS/Lesson16/salih/script.js

```diff
@@ -116,27 +116,91 @@ for(const key in obj) {
 console.log(sum);
 console.log('- - - - - - - - - - - - - - - - - ');
 // 12. Write a loop that finds the average of an array of numbers.
-const numbers12 = [10, 20, 30, 40, 50];
+const numbers12 = [10, 20, 30, 40, 500];
+function sumArray(numbers){
+	let sum=0;
+	for (i=0; i< numbers.length; i++){
+		sum += numbers[i];
+	};
+	return sum/numbers.length;
+}
+console.log('Exercise 12.1: ', sumArray(numbers12));
 
 console.log('- - - - - - - - - - - - - - - - - ');
 // 13. Write a loop that prints out the sum of all even numbers between 1 and 50.
+function sumEvenNumbers(firstNum, secondNum){
+	let small, big, sum=0;
+	if (firstNum > secondNum){ 
+		small = secondNum;
+		big = firstNum;
+	} else {
+		small = firstNum;
+		big = secondNum;
+	};
+	console.log('11-The Small Number is ', small, '. The Big Number is ', big); //I used them to follow the code.
+	for (i=small; i < big+1; i++){
+		console.log('22-The Small Number is ', small, '. The Big Number is ', big, 'The i Number is ', i); //I used them to follow the code.
+		if (i % 2 == 0){
+			sum += small;
+			console.log('33-The sum is ', sum); //I used them to follow the code.
+		};
+	};
+	return sum;
+}
 
-
-
+console.log('Exercise 13.1: ', sumEvenNumbers(1, 50));
 console.log('- - - - - - - - - - - - - - - - - ');
 // 14. Write a loop that finds the second largest number in an array of numbers.
-const numbers14 = [10, 5, 20, 8, 15];
+const numbers14 = [10, 5, 20, 8, 15, 42];
+function orderingSmalToBig(numbers){
```

Comment:
I'd name this function `findSecontLargestNumber` as it doesn't order anything :)

Reply:
You solution works but there are other solutions that could be easier to read. I think the most common approach to solve this problem is the following:
- compare the first element of the array (array[0]) and the second (array[1]). Assign the biggest value to the variable named largestNumber, and the other value to the secondLargest.
- loop through the ret of the array (starting with array[2]) and compare each element to the largestNumber and secondLargest, if the new value is bigger than any of them, update largestNumber and secondLargest accordingly.